deepspeed/launcher/launch.py

```diff
@@ -211,8 +284,42 @@ def main():
 
             # spawn the processes
             cmd = []
+            if args.bind_cores_to_rank:
+                if 'KMP_AFFINITY' in os.environ.keys():
```

Comment:
Can we add a check to ensure `numactl` is installed here? then we can exit gracefully telling the user they need this as a dependency if args like `bing_cores_to_rank` are requested.

Reply:
Added a check for `numactl` in the begining of `bind_cores_to_ranks` and exit with proper comment.
https://github.com/microsoft/DeepSpeed/pull/2881/commits/f72d57678c37a233edc29bd1321d04ba01952230

Reply:
@delock, this looks great. Can you make the error message actionable for users by including a way for them to install the missing package. This could be similar to: 
https://github.com/microsoft/DeepSpeed/blob/5f5cc82415b11ab2c9bf85969deb510fe8631446/op_builder/async_io.py#L72

Reply:
Availablility check had been made according to libaio example. The new check for numactl will be a 'soft' one since the check is only for popular distros.

Reply:
Below is the error message on my environment when numactl is not installed.
```
please install the numactl package with apt
Traceback (most recent call last):
```